### PR TITLE
[C++] AVRO-4058: Allow custom attributes in arrays

### PR DESCRIPTION
**Pull request by pascalginter**
[AVRO-4058](https://issues.apache.org/jira/browse/AVRO-4058)

## What is the purpose of the change
Other languages (i.e. Java) allow adding custom attributes in the schema of an array while the C++ implementation does not. The sparse spec states to allow all custom attributes ("Attributes not defined in this document are permitted as metadata, but must not affect the format of serialized data") and other projects using avro (i.e. Iceberg) depend on custom attributes in arrays.

This PR allows adding custom attributes to arrays.

## Verifying this change

This change added tests and can be verified as follows:

Run SchemaTests, I added a single additional test case in testRoundTrip 

## Documentation

- Does this pull request introduce a new feature?  no
- If yes, how is the feature documented? not applicable

### CODE REVIEW

**Review by martin-g — APPROVED** [review]
LGTM!

**Review by mkmkme — APPROVED** [review]

**Review by martin-g — APPROVED** [review]

File: lang/c++/test/SchemaTests.cc [hunk 1]
```diff
@@ -355,7 +355,8 @@ const char *roundTripSchemas[] = {
             {"name":"f1","type":"long","extra_field":"1"},
             {"name":"f2","type":"int","extra_field1":"21","extra_field2":"22"}
         ]
-    })"
+    })",
+    R"({"type":"array","items":"long","extra":"1"})"
```
Comment (martin-g) [inline lang/c++/test/SchemaTests.cc:359]: There is an error:
```
unknown location(0): fatal error: in "Avro C__ unit tests for schemas/avro__schema__testBasic_35": avro::Exception: Invalid type. Expected "string" actual long
/home/runner/work/avro/avro/lang/c++/test/SchemaTests.cc(428): last checkpoint: {"type": "array", "items": "long", "extra attribute": 1}
```
Where does `attribute` come from in `"extra attribute"` ?!?

Comment (pascalginter) [reply lang/c++/test/SchemaTests.cc:359]: The issue is not in the added test case but in an existing test case (line 146) in testBasic checking that custom attributes do not impact the ability to compile schemas.
The specified attribute in the test was given as a integer, which so far was no issue as it was ignored but now it is attempted to be parsed.  This attempt failed as only strings are supported in custom attributes. For now I changed the attribute to a string, the original test case can be restored once non-string attributes are supported.
(This change of 1 to "1" was present in the original PR, I accidentally undid it when merging the latest main)

**Comment by martin-g** [comment]
Thank you, @pascalginter !

**Pull request merged**